earth2studio/models/px/dlesym.py

```diff
@@ -0,0 +1,1077 @@
+        times_flattened = np.array(
+            [[a + t for t in timedeltas] for a in anchor_times]
+        ).flatten()
+        sol = insolation(
+            times_flattened,
+            self.hpx_lat,
+            self.hpx_lon,
+        )
+        t, f, h, w = sol.shape
+        sol = torch.from_numpy(sol).view(len(anchor_times), len(timedeltas), 1, f, h, w)
+        sol = sol.permute(0, 3, 1, 2, 4, 5)
+        return sol
+
+    def _make_atmos_coupling(
+        self, x: torch.Tensor, coords: CoordSystem
+    ) -> torch.Tensor:
+        """Make atmos coupling tensor from input data
+        Assumes `x` is a tensor of shape (batch, lead_time, variable, face, height, width).
+        Also assumes `x` has been properly normalized.
+
+        Parameters
+        ----------
+        x : torch.Tensor
+            Input data
+        coords : CoordSystem
+            Input coordinates
+
+        Returns
+        -------
+        torch.Tensor
+            Atmos coupling tensor: shape (lead_time, batch, variable, face, height, width)
+        """
+
+        atmos_coupling_channel_idx = [
+            list(coords["variable"]).index(var) for var in self.atmos_coupling_variables
+        ]
+        atmos_coupling_lead_time_idx = [
+            list(coords["lead_time"]).index(t)
+            for t in self.atmos_coupling_times + coords["lead_time"][-1]
+        ]
+        atmos_coupling = x[:, atmos_coupling_lead_time_idx][
+            ..., atmos_coupling_channel_idx, :, :, :
+        ].permute(1, 0, 2, 3, 4, 5)
+
+        return atmos_coupling
+
+    def _make_ocean_coupling(
+        self, x: torch.Tensor, coords: CoordSystem
+    ) -> torch.Tensor:
+        """Make ocean coupling tensor from atmos outputs
+        Assumes `x` is a tensor of shape (batch, face, lead_time, variable, height, width).
+        Also assumes `x` has been properly normalized.
+
+        Parameters
+        ----------
+        x : torch.Tensor
+            Input data
+        coords : CoordSystem
+            Input coordinates
+
+        Returns
+        -------
+        torch.Tensor
+            Ocean coupling tensor: shape (lead_time, batch, variable, face, height, width)
+        """
+
+        # Subselect the ocean coupling variables -- we use all atmos lead times here so we can average over them
+        ocean_coupling_channel_idx = [
+            list(coords["variable"]).index(var) for var in self.ocean_coupling_variables
+        ]
+        ocean_coupling = x[:, :, :, ocean_coupling_channel_idx, :, :]
+
+        # Slice along the lead_time dim, average, and stack along the variable dim
+        # (different time-averaged quantities are defined as different variables)
+        slices = ocean_coupling.chunk(len(self.ocean_input_times), dim=2)
+        ocean_coupling = torch.concat(
+            [s.mean(dim=2, keepdim=True) for s in slices], dim=3
+        )
+        ocean_coupling = ocean_coupling.permute(2, 0, 3, 1, 4, 5)
```

Comment:
why is this different than the _make_atmos_coupling, do the models have different input dimensions?

Reply:
The ocean coupling inputs are the outputs of the atmosphere model (for lead times +6h, ..., +96h) with time-averaging applied to them. There is no time averaging for the atmos coupling inputs.